src/createServer.js

```diff
@@ -1,9 +1,27 @@
 /* eslint-disable no-console */
 'use strict';
 
+const http = require('http');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  return http.createServer((req, res) => {
+    const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
```

Comment:
The URL normalization assumes that the request URL is always absolute by using `req.headers.host`. This might cause issues if the `host` header is missing or malformed. Consider adding error handling for cases where `req.headers.host` is not present or invalid.